implementations/rust/ockam/ockam_entity/src/lib.rs

```diff
@@ -8,39 +8,86 @@
     unsafe_code,
     unused_import_braces,
     unused_qualifications,
-    // TODO re-enable warnings
+    // warnings
 )]
 
-use crate::EntityError;
+use std::sync::{Arc, Mutex};
+
+pub use change::*;
+pub use channel::*;
+pub use contact::*;
+pub use entity::*;
+pub use error::*;
+pub use identifiers::*;
+pub use key_attributes::*;
 use ockam_channel::SecureChannelVault;
-use ockam_core::lib::HashMap;
-use ockam_core::{Address, Result};
-use ockam_node::Context;
-use ockam_vault_core::{Hasher, KeyIdVault, SecretVault, Signer, Verifier};
-use ockam_vault_sync_core::VaultSync;
+use ockam_core::{lib::HashMap, Address, Message, Result};
+use ockam_node::{block_future, Context};
+use ockam_vault::{Hasher, KeyIdVault, SecretVault, Signer, Verifier};
+pub use profile_state::*;
+pub use traits::*;
+pub use worker::*;
+
+use crate::EntityError;
+
+pub struct ContextAddress {
+    ctx: Context,
+    address: Address,
+}
+
+pub type Handle = Arc<Mutex<ContextAddress>>;
+
+impl ContextAddress {
+    pub fn new(ctx: Context, address: Address) -> Self {
+        ContextAddress { ctx, address }
+    }
+
+    pub async fn async_cast<M: Message + Send + 'static>(&mut self, msg: M) -> Result<()> {
+        self.ctx.send(self.address.clone(), msg).await
+    }
+
+    pub fn cast<M: Message + Send + 'static>(&mut self, msg: M) -> Result<()> {
+        block_future(
+            &self.ctx.runtime(),
+            async move { self.async_cast(msg).await },
+        )
+    }
+
+    pub async fn async_call<I: Message + Send + 'static, O: Message + Send + 'static>(
+        &mut self,
+        msg: I,
+    ) -> Result<O> {
+        self.ctx.send(self.address.clone(), msg).await?;
```

Comment:
There is probability we need to spawn new ctx here each time

Reply:
Interesting, will look at it

Reply:
Implemented